metadata-ingestion/src/datahub/api/entities/datajob/dataflow.py

```diff
@@ -56,21 +69,24 @@ class DataFlow:
     env: Optional[str] = None
 
     def __post_init__(self):
-        if self.env is not None:
-            self.urn = DataFlowUrn.create_from_ids(
-                orchestrator=self.orchestrator,
-                env=self.env,
-                flow_id=self.id,
-                platform_instance=self.platform_instance,
-            )
-        else:
-            self.urn = DataFlowUrn.create_from_ids(
-                orchestrator=self.orchestrator,
-                env=self.cluster,
-                flow_id=self.id,
-                platform_instance=self.platform_instance,
+        if self.env is not None and self.cluster is not None:
+            raise ValueError(
+                "Cannot provide env and cluster parameter both at the same time. cluster is deprecated use env."
             )
 
+        if self.env is None and self.cluster is None:
+            raise ValueError("env is required parameter.")
+
+        if self.env is None and self.cluster is not None:
+            logger.warning("Use env argument. The cluster argument is deprecated")
```

Comment:
```suggestion
            logger.warning("The cluster argument is deprecated. Use env and possibly platform_instance instead.")
```

Reply:
also just do `self.env = self.cluster`

then you can remove the `if self.env is not None` logic below, and get rid of the casts in `dataprocess_instance.py`

Reply:
self.env is optional, cast is required to pass lint

Reply:
self.env is optional, cast is required to pass lint

Reply:
self.env is optional, cast is required to pass lint